src/modules/filters/components/FilterBar/FilterBar.jsx

```diff
@@ -12,74 +13,68 @@ const { Option } = Select;
 const FilterBar = () => {
     const { styles } = useFelaEnhanced(felaRules);
     const intl = useIntl();
+
+    const { fetchReports, inProgress } = useReportsFetcher();
     const { appName, platform, deviceModel } = useFilters();
+
+    const [form] = Form.useForm();
+
+    const handleChange = form => {
+        fetchReports(form.getFieldsValue());
+    };
+    const filterProps = [
+        {
+            name: 'appName',
+            options: ['All'].concat(appName),
+            label: 'App name',
+        },
+        {
+            name: 'platform',
+            options: ['All'].concat(platform),
+            label: 'Platform',
+        },
+        {
+            name: 'deviceModel',
+            options: ['All'].concat(deviceModel),
+            label: 'Device model',
+        },
+    ];
+
     return (
         <div className={styles.container}>
-            <Form className={styles.container} size="large" layout="vertical">
+            <Form
+                className={styles.container}
+                onFieldsChange={() => handleChange(form)}
+                form={form}
+                initialValues={{ appName: 'All', deviceModel: 'All', platform: 'All' }}
```

Comment:
this All should be constant and also translated

Reply:
done